.github/workflows/machine-os-pr.yml

```diff
@@ -0,0 +1,145 @@
+name: "Machine OS PR"
+
+on:
+  pull_request_target:
+    # prevent action from running on older release-process branches
+    # TODO: remove when we move to new release flow
+    branches:
+      - 'v5.5'
+      - 'main'
+    paths:
+      - 'version/rawversion/version.go'
+
+concurrency:
+  # Cancel other in-progress runs on re-pushes
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  podman-image-build-pr:
+    name: Open PR on podman-machine-os
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      issues: write
+      pull-requests: write
+    env:
+      SHA: ${{github.event.pull_request.head.sha}}
+      UPSTREAM_MACHINE_OS: "containers/podman-machine-os"
+      PODMAN_REPO: "containers/podman"
+    steps:
+
+      - name: Get version
+        id: getversion
+        run: |
+          VERSION=$(curl "https://raw.githubusercontent.com/$PODMAN_REPO/$SHA/version/rawversion/version.go" | sed -n 's/^const RawVersion = \"\(.*\)\"$/\1/p')
+          # ignore -dev version bumps unless on main
+          if [[ ${{github.base_ref}} != "main" ]] && [[ $VERSION == *-dev ]] ; then
+            echo "::warning:: SKIPPING: dev bump not on main"
+          elif [[ ${{github.base_ref}} == *-rhel ]] ; then
+            echo "::warning:: SKIPPING: rhel branch"
+          else
+            echo "update=true" >> "$GITHUB_OUTPUT"
+          fi
+          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+
+      - name: Check machine-os-branch
+        if: steps.getversion.outputs.update == 'true'
+        run: |
+          if ! (curl -s https://api.github.com/repos/$UPSTREAM_MACHINE_OS/branches| jq -e --arg branch "${{github.base_ref}}" '.[] | select(.name==$branch)') ; then
+            echo "::error:: Release branch does not exist."
+            echo "::error:: Please push $branch to $UPSTREAM_MACHINE_OS, then re-run this task."
+            exit 1
```

Comment:
sounds fine for now, I guess long term it would be nice if it can create the branch. But that would also mean the bot account needs write access on the repo which means it could be an attack target so maybe it is fine if we do that manually.